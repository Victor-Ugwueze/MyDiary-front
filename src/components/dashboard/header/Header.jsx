// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const Header = ({ onClick, user }) => (
  <header className="dashboard-header">
      <img className="loading-indicator" src="/images/loading_spinner.gif" />
    <span className="icon-bar menu-icon" onClick={onClick}></span>
    <div id="top-bar">
      <div className="brand">
          <img src="/images/logo.png" id="logo" />
      </div>
      <nav>
          <ul className="navbar">
            <div id="notification-icon">
              <img src="/images/icons/notification_white_1.png" />
              { /* <span id="number-of-notifications">7</span> */ }
            </div>
              <li className="nav-link">
                  <div className="dropdown">
                    <div className="user-img">
                        <img className="profile-img" src="/images/avatar.png" alt="" />
                        <span id="menu-firstname">{user.firstName}</span>
                    </div>
                  </div>
              </li>
          </ul>
      </nav>
    </div>
  </header>
);

Header.propTypes = {
  onClick: PropTypes.func.isRequired,
  user: PropTypes.object
};

export default Header;
