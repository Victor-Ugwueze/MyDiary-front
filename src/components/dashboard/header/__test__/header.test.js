// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import Header from '../Header';


configure({
  adapter: new Adapter(),
});

describe('Renders modals', () => {
  it('should render the header component', () => {
    const showSideBar = () => { };

    const component = shallow(
      <Header onClick={showSideBar} user={{ firstName: 'victor' }}/>
    );
    expect(component).toMatchSnapshot();
  });
});
