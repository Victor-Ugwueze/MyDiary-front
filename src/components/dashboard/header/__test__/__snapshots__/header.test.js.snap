// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders modals should render the header component 1`] = `
<header
  className="dashboard-header"
>
  <img
    className="loading-indicator"
    src="/images/loading_spinner.gif"
  />
  <span
    className="icon-bar menu-icon"
    onClick={[Function]}
  />
  <div
    id="top-bar"
  >
    <div
      className="brand"
    >
      <img
        id="logo"
        src="/images/logo.png"
      />
    </div>
    <nav>
      <ul
        className="navbar"
      >
        <div
          id="notification-icon"
        >
          <img
            src="/images/icons/notification_white_1.png"
          />
        </div>
        <li
          className="nav-link"
        >
          <div
            className="dropdown"
          >
            <div
              className="user-img"
            >
              <img
                alt=""
                className="profile-img"
                src="/images/avatar.png"
              />
              <span
                id="menu-firstname"
              />
            </div>
          </div>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
